### ISSUE TITLE
Vagrant internal network config

### ISSUE THREAD

**Issue by skhoroshavin** [metacloud/molecule#1421]
# Issue Type

- Feature request

# Desired Behavior

When using Vagrant driver with VirtualBox provider it is sometimes desirable to create machines with network type "internal". In plain Vagrant it is done by providing additional parameter
```
config.vm.network "private_network", ip: "192.168.50.4", virtualbox__intnet: "mynetwork"
```
While it can be done by using `instance_raw_config_args` it has some limitations. For example it prevents using YAML anchors for pasting specific IPs. Also it would be much more convenient to use standard `interfaces` section. I've looked through code and it seems there are two options:
- instead of hardcoding parameters passed based on `type` property just pass them all
- add one more if-else branch with another `type` option

Please let me know whether you are going to accept PR with this functionality and if yes what implementation would you prefer.

**Comment by retr0h** [comment]
Sounds fine to me.

**Comment by skhoroshavin** [comment]
Implemented in PR #1422

**Comment by retr0h** [comment]
merged, thanks @skhoroshavin \m/
